Inductor cpp wrapper: fix cumsum codegen

chunyuan-w opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #116171

Fixes https://github.com/pytorch/pytorch/issues/115829

For `cumsum(Tensor self, int dim, *, ScalarType? dtype=None) -> Tensor`, `dim` is not a `kwarg_only` argument, but it could be provided as a kwarg when calling this OP.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

chunyuan-w committed:
Inductor cpp wrapper: fix cumsum codegen

[ghstack-poisoned]

chunyuan-w commented:
@pytorchbot rebase

chunyuan-w committed:
Update on "Inductor cpp wrapper: fix cumsum codegen"


Fixes https://github.com/pytorch/pytorch/issues/115829

For `cumsum(Tensor self, int dim, *, ScalarType? dtype=None) -> Tensor`, `dim` is not a `kwarg_only` argument, but it could be provided as a kwarg when calling this OP.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

chunyuan-w committed:
Update on "Inductor cpp wrapper: fix cumsum codegen"


Fixes https://github.com/pytorch/pytorch/issues/115829

For `cumsum(Tensor self, int dim, *, ScalarType? dtype=None) -> Tensor`, `dim` is not a `kwarg_only` argument, but it could be provided as a kwarg when calling this OP.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

desertfire commented on torch/_inductor/ir.py:
```diff
@@ -4401,6 +4401,7 @@ class ExternKernelNode:
 
 class FallbackKernel(ExternKernelAlloc):
     args_default_value: List[Dict[str, Any]]
+    args_name: List[str]
```
Can we just extend `args_default_value` to include names instead of adding another class member?

chunyuan-w committed:
Update on "Inductor cpp wrapper: fix cumsum codegen"


Fixes https://github.com/pytorch/pytorch/issues/115829

For `cumsum(Tensor self, int dim, *, ScalarType? dtype=None) -> Tensor`, `dim` is not a `kwarg_only` argument, but it could be provided as a kwarg when calling this OP.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

desertfire commented on torch/_inductor/ir.py:
```diff
@@ -4678,9 +4686,13 @@ def codegen(self, wrapper):
+                            "kwarg_only": x.kwarg_only,
+                        }
                         for x in schema.arguments
-                        if not x.kwarg_only
```
I guess you had to drop this `if not x.kwarg_only` to make it work?

desertfire replied:
No, I don't think it should be removed.

chunyuan-w replied:
On second thought, there's no need to remove it. I've updated the code.

chunyuan-w committed:
Update on "Inductor cpp wrapper: fix cumsum codegen"


Fixes https://github.com/pytorch/pytorch/issues/115829

For `cumsum(Tensor self, int dim, *, ScalarType? dtype=None) -> Tensor`, `dim` is not a `kwarg_only` argument, but it could be provided as a kwarg when calling this OP.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

chunyuan-w commented:
@pytorchbot merge
